main.py

```diff
@@ -19,22 +18,153 @@ def get_horsepower(self):
-    assert vehicle2.get_classification == 'Midsize'
+    assert vehicle2.get_classification == 'Midsize Vehicle'
     vehicle3 = Vehicle(200)
     assert vehicle3.get_horsepower == 200
-    assert vehicle3.get_classification == 'Fullsize'
+    assert vehicle3.get_classification == 'Fullsize Vehicle'
+
+def test_car():
+    '''Test car class'''
+    car1 = Car(100, 2)
+    assert car1.get_horsepower() == 100
+    assert car1.get_classification() == 'Compact Coupe'
+    car2 = Car(100, 4)
+    assert car2.get_horsepower() == 100
+    assert car2.get_classification() == 'Compact Sedan'
+    car3 = Car(150, 2)
+    assert car3.get_horsepower() == 150
+    assert car3.get_classification() == 'Midsize Coupe'
+    car4 = Car(150, 4)
+    assert car4.get_horsepower() == 150
+    assert car4.get_classification() == 'Midsize Sedan'
+    car5 = Car(200, 2)
+    assert car5.get_horsepower() == 200
+    assert car5.get_classification() == 'Fullsize Coupe'
+    car6 = Car(200, 4)
+    assert car6.get_horsepower() == 200
+    assert car6.get_classification() == 'Fullsize Sedan'
+
+def test_truck():
+    '''Test Truck class'''
+    truck1 = Truck(100, 'RWD')
+    assert truck1.get_horsepower() == 100
+    assert truck1.get_classification() == 'Compact RWD Truck'
+    truck2 = Truck(100, '4x4')
+    assert truck2.get_horsepower() == 100
+    assert truck2.get_classification() == 'Compact 4x4 Truck'
+    truck3 = Truck(150, 'RWD')
+    assert truck3.get_horsepower() == 150
+    assert truck3.get_classification() == 'Midsize RWD Truck'
+    truck4 = Truck(150, '4x4')
+    assert truck4.get_horsepower() == 150
+    assert truck4.get_classification() == 'Midsize 4x4 Truck'
+    truck5 = Truck(200, 'RWD')
+    assert truck5.get_horsepower() == 200
+    assert truck5.get_classification() == 'Fullsize RWD Truck'
+    truck6 = Truck(200, '4x4')
+    assert truck6.get_horsepower() == 200
+    assert truck6.get_classification() == 'Fullsize 4x4 Truck'
+
+def test_vehicle_dunder():
+    '''Test Vehicle dunder methods'''
+    vehicle1 = Vehicle(100)
+    vehicle2 = Vehicle(150)
+    vehicle3 = Vehicle(200)
+    assert repr(vehicle1) == 'Vehicle with 100 horsepower'
+    assert repr(vehicle2) == 'Vehicle with 150 horsepower'
+    assert repr(vehicle3) == 'Vehicle with 200 horsepower'
+    assert str(vehicle1) == 'Vehicle with 100 horsepower'
+    assert str(vehicle2) == 'Vehicle with 150 horsepower'
+    assert str(vehicle3) == 'Vehicle with 200 horsepower'
+    assert vehicle1 != vehicle2
+    assert vehicle1 != vehicle3
+    assert vehicle2 != vehicle1
+    assert vehicle2 != vehicle1
+    assert vehicle3 != vehicle1
+    assert vehicle3 != vehicle2
+    assert vehicle1 < vehicle2
+    assert vehicle1 < vehicle3
+    assert vehicle2 > vehicle1
+    assert vehicle2 < vehicle3
+    assert vehicle3 > vehicle1
+    assert vehicle3 > vehicle2
 
 if __name__ == 'main':
     test_vehicle()
+    test_car()
+    test_truck()
+    test_vehicle_dunder()
```

Comment:
# Week 10 Grading

| Points | Location        | Criteria                                |
| ------ | --------------- | --------------------------------------- |
| 5      | main.py         | `__repr__` dunder method                |
| 5      | main.py         | `__str__` dunder method                 |
| 5      | main.py         | `__eq__` dunder method                  |
| 5      | main.py         | `__lt__` dunder method                  |
| 5      | main.py         | `__gt__` dunder method                  |
| 5      | pytest coverage | Tests cover all aspects of main.py      |
| 10     | pylint          | Pylint score >= 8.0.                    |
|  -1      | instructor      | Code style, efficiency, and readability |
| 39/40 |